pkg/util/tracing/crdbspan.go

```diff
@@ -543,8 +543,13 @@ func (s *crdbSpan) recordFinishedChildrenLocked(childRecording tracingpb.Recordi
 		}
 
 		// We don't have space for this recording. Let's collect just the structured
-		// records by falling through.
-		fallthrough
+		// records.
+		for ci := range childRecording {
```

Comment:
hmm was there another bug hiding in that we only collected the structured events from the root of a verbose recording if we ran out of space. When the comment indicates that we want to collect events from the whole recording. If thats the case do you think you can add this to the regression test where each child span can:

`child.RecordStructured(&types.Int32Value{Value: 4})`

and then right at the end we can `sp.FinishAndGetConfiguredRecording()` and check we have the right number of structured events?

Reply:
Good idea, done.